File: Python/Estoque.py
produtos = []

def verificar_estoque_critico(quantidade):
    if quantidade < 5:
        return " [REPOSIÇÃO NECESSÁRIA]"
    return ""

def exibir_cabecalho():
    print("=" * 35)
    print("      SORVETERIA DO CACA 🍦")
    print("=" * 35)

while True:
    exibir_cabecalho()
    print("\n--- Gerenciador de Estoque ---")
    print("1 - Adicionar novo produto")
    print("2 - Listar produtos")
    print("3 - Fechamento do estoque")
    print("4 - Sair")

    opcao = input("Escolha uma opção: ")

    if opcao == "1":
        nome = input("Nome do produto: ")

        try:
            quantidade = int(input("Quantidade em estoque: "))
            preco = float(input("Preço unitário: "))
        except ValueError:
            print("Erro: digite valores válidos!")
            continue

        if quantidade < 0:
            print("Erro: a quantidade não pode ser negativa!")
        else:
            produto = {
                "nome": nome,
                "quantidade": quantidade,
                "preco": preco
            }
            produtos.append(produto)
            print("Produto cadastrado com sucesso ✅")

    elif opcao == "2":
        if len(produtos) == 0:
            print("Nenhum produto cadastrado.")
        else:
            print("\n--- Produtos em Estoque ---")
            for p in produtos:
                alerta = verificar_estoque_critico(p["quantidade"])

                print(f"Produto: {p['nome']}{alerta}")
                print(f"Quantidade: {p['quantidade']}")
                print(f"Preço: R$ {p['preco']:.2f}")
                print("-" * 30)

    elif opcao == "3":
        print("\n--- FECHAMENTO DO ESTOQUE ---")

        if len(produtos) == 0:
            print("Nenhum produto cadastrado.")
        else:
            total_geral = 0

            for p in produtos:
                subtotal = p["quantidade"] * p["preco"]
                total_geral += subtotal

                alerta = verificar_estoque_critico(p["quantidade"])

                print(f"{p['nome']}{alerta}")
                print(f"Valor: R$ {subtotal:.2f}")

            print(f"\nValor total em estoque: R$ {total_geral:.2f}")

    elif opcao == "4":
        print("Encerrando o sistema... 👋")
        break

    else:
        print("Opção inválida!")

File: Java/Estoque.java
import java.util.ArrayList;
import java.util.Scanner;

class Produto {
    String nome;
    int quantidade;
    double preco;

    Produto(String nome, int quantidade, double preco) {
        this.nome = nome;
        this.quantidade = quantidade;
        this.preco = preco;
    }
}

public class Estoque {

    public static String verificarEstoqueCritico(int quantidade) {
        if (quantidade < 5) {
            return " [REPOSIÇÃO NECESSÁRIA]";
        }
        return "";
    }

    public static void exibirCabecalho() {
        System.out.println("==================================");
        System.out.println("        SORVETERIA DO CACA 🍦");
        System.out.println("==================================");
    }

    public static void main(String[] args) {

        try (Scanner sc = new Scanner(System.in)) {
            ArrayList<Produto> produtos = new ArrayList<>();

            while (true) {
                exibirCabecalho();

                System.out.println("\n--- Gerenciador de Estoque ---");
                System.out.println("1 - Adicionar novo produto");
                System.out.println("2 - Listar produtos");
                System.out.println("3 - Fechamento do estoque");
                System.out.println("4 - Sair");

                System.out.print("Escolha uma opção: ");
                String opcao = sc.nextLine();

                switch (opcao) {
                    case "1":
                        try {
                            System.out.print("Nome do produto: ");
                            String nome = sc.nextLine();

                            System.out.print("Quantidade em estoque: ");
                            int quantidade = Integer.parseInt(sc.nextLine());

                            System.out.print("Preço unitário: ");
                            double preco = Double.parseDouble(sc.nextLine());

                            if (quantidade < 0) {
                                System.out.println("Erro: quantidade não pode ser negativa!");
                            } else {
                                produtos.add(new Produto(nome, quantidade, preco));
                                System.out.println("Produto cadastrado com sucesso ✅");
                            }

                        } catch (Exception e) {
                            System.out.println("Erro: digite valores válidos!");
                        }
                        break;

                    case "2":
                        if (produtos.isEmpty()) {
                            System.out.println("Nenhum produto cadastrado.");
                        } else {
                            for (Produto p : produtos) {
                                String alerta = verificarEstoqueCritico(p.quantidade);

                                System.out.println("\nProduto: " + p.nome + alerta);
                                System.out.println("Quantidade: " + p.quantidade);
                                System.out.printf("Preço: R$ %.2f\n", p.preco);
                            }
                        }
                        break;

                    case "3":
                        double total = 0;

                        for (Produto p : produtos) {
                            double subtotal = p.quantidade * p.preco;
                            total += subtotal;

                            String alerta = verificarEstoqueCritico(p.quantidade);

                            System.out.println(p.nome + alerta);
                            System.out.printf("Valor: R$ %.2f\n", subtotal);
                        }

                        System.out.printf("\nValor total em estoque: R$ %.2f\n", total);
                        break;

                    case "4":
                        System.out.println("Encerrando o sistema... 👋");
                        return;

                    default:
                        System.out.println("Opção inválida!");
                }
            }
        }
    }
}
